src/app/progress-bar/progress-bar.component.ts

```diff
@@ -0,0 +1,48 @@
+import { Component, OnInit } from '@angular/core';
+
+import { SoldProducts } from '../product';
+import { ProductService } from '../product.service';
+
+@Component({
+  selector: 'app-progress-bar',
+  templateUrl: './progress-bar.component.html',
+  styleUrls: ['./progress-bar.component.scss']
+})
+export class ProgressBarComponent implements OnInit {
+  progressValue: number = 0;
+
+  constructor(private productService: ProductService) { }
+
+  ngOnInit(): void {
+    this.getSoldProducts();
+  }
+
+  processData(soldProducts: SoldProducts):void {
+    let total: number = 0 ;
+    soldProducts.totalValue;
+    soldProducts.data.forEach( product => {
+      return total += product.value;
+    });
+
+    this.showProgress(this.calculateProgress(total, soldProducts.totalValue));
+  }
+
+  getSoldProducts(): void {
+    this.productService.getSoldProducts()
+      .subscribe(val => this.processData(val));
```

Comment:
Never subscribe inside of method. This is creating sideffect and polite your code. This way you cannot clean after your self with unsubscribe. Try to design code that way to have only async pipe or just single subscription in onInit with takeUntil.

Reply:
Removed .subscribe, used async pipe